### PR TITLE
fix: Update docs and use `dotenvy` instead of `dotenv`

### LINKED ISSUE #380: dotenv is unmaintained, we should use dotenvy.
We should migrate to dotenvy

### PR DESCRIPTION
**Pull request by MaanavKhaitan**
# What

Based on Avi's feedback:

- Uses `dotenvy` instead of `dotenv` closes #380 
- Points to `CONTRIBUTING.md` from the book
- Adds brief info about public/private inputs and shortening the fraud proof window by submitting a ZK validity proof

# Testing

`just test-all`: Passes

### CODE REVIEW

File: book/src/integration/using-program.md [hunk 1]
```diff
@@ -26,3 +26,6 @@ Offchain job requests can be sent by a user, your app, or any authorized third-p
 
 You can read more in the [Offchain Jobs](./offchain.md) section.
 
+### Public and private inputs
+
+For now, all inputs are public. We are working on adding support for private inputs.
```
Comment (emostov) [inline book/src/integration/using-program.md:31]: nit: we aren't working on it per se but maybe  `We plan on adding support for private inputs in the future.`

**Review by emostov — APPROVED** [review]
LGTM - one non essential nit

File: book/src/integration/using-program.md (hunk 1 above)
Comment (lucasege) [reply book/src/integration/using-program.md:31]: nit: it seems like adding support for private inputs would be incompatible with the current IVM design. so yeah i agree "plannign to eventually maybe add support" is probably correct

Comment (MaanavKhaitan) [reply book/src/integration/using-program.md]: updated

**Review by Avi-D-coder — APPROVED** [review]

**Review by lucasege — APPROVED** [review]

**Pull request merged**